build/common/installer/scripts/fluent-bit-conf-customizer.rb

```diff
@@ -84,6 +85,12 @@ def substituteFluentBitPlaceHolders
       new_contents = new_contents.gsub("#${KubernetesFilterEnabled}", "")
     end
 
+    if !annotationBasedLogFiltering.nil? && annotationBasedLogFiltering.to_s.downcase == "true"
```

Comment:
same do we need to add condition for KubernetesFilterEnabled

Reply:
no. here's why. if KubernetesFilterEnabled is enabled then it will replace all the comments from fluent-bit.conf and therefore line 90 will be a noop

if KubernetesFilterEnabled  is false then line 90 will remove the comments but since KubernetesFilterEnabled  is disabled oms.go will not process any metadata

this way filtering will work without doing any additional metadata related tasks